### DISCUSSION TITLE
New CI testing for pyrad

### DISCUSSION THREAD

**Discussion by wolfidan** [MeteoSwiss/pyrad#42] [Announcements]
Dear all, dear @jfigui ,

I have setup new CI routines to test pyrad. They rely on this new repository
https://github.com/MeteoSwiss/pyrad-tests

A set of config files are defined in this repo, divided in two categories, that correspond to two different tests:

- *base* : these are basic routines that work with the standard version of Py-ART (https://github.com/ARM-DOE/pyart)
- *mch* : these are more sophisticated routines that work only with the MeteoSwiss version of Py-ART (https://github.com/MeteoSwiss/pyart).

On the README of this repo there are guidelines on how to create new tests.
The CI works in the following way: all tests are run sequentially. For every test, the created outputs are compared with the reference outputs in the pyrad-tests repository. If all outputs match, the test is successful. Currently file comparisons are run only for csv and netCDF files. 

**No file testing is currently done on png files because the content of png files changes significantly between different OS. It was impossible to get exactly the same image on the github worker as on the server which was used to create the references.**

Until we find a fix for this, I would suggest to add only tests that generate csv files or netCDF files, because png files are ignored.

**Comment by jfigui** [comment]
Brilliant @wolfidan !

I would say what really matters is that data files are generated properly. We know that images are difficult to replicate 100%
